universe/node/api_mutual_docks.go

```diff
@@ -222,21 +232,33 @@ func createWorldPortal(portalName string, from, to universe.World) (uuid.UUID, e
 		return uuid.Nil, errors.WithMessage(err, "failed to get portal space type id")
 	}
 
-	template := helper.SpaceTemplate{
-		SpaceName:   &portalName,
-		SpaceTypeID: portalSpaceTypeID,
-		ParentID:    dockingStation.GetID(),
-		SpaceAttributes: []*entry.Attribute{
-			entry.NewAttribute(
-				entry.NewAttributeID(universe.GetSystemPluginID(), universe.ReservedAttributes.World.TeleportDestination.Name),
-				entry.NewAttributePayload(
-					&entry.AttributeValue{
-						universe.ReservedAttributes.World.TeleportDestination.Key: to.GetID().String(),
-					},
-					nil,
-				),
+	spaceAttributes = append(spaceAttributes, entry.NewAttribute(
+		entry.NewAttributeID(universe.GetSystemPluginID(), universe.ReservedAttributes.World.TeleportDestination.Name),
+		entry.NewAttributePayload(
+			&entry.AttributeValue{
+				universe.ReservedAttributes.World.TeleportDestination.Key: to.GetID().String(),
+			},
+			nil,
+		),
+	))
+
+	if portalImage != "" {
+		spaceAttributes = append(spaceAttributes, entry.NewAttribute(
+			entry.NewAttributeID(universe.GetSystemPluginID(), universe.ReservedAttributes.Space.PortalDockFace.Name),
+			entry.NewAttributePayload(
+				&entry.AttributeValue{
+					universe.ReservedAttributes.Space.PortalDockFace.Key: portalImage,
```

Comment:
Maybe rename `PortalDockFace` to `PortalDockImage` or Thumbnail ?

Reply:
i will keep it this way since the space_attribute is called dock_face as well. this solution should also not be a permanent one